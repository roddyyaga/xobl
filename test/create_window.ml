let () =
  Lwt_main.run
    (let%lwt resp, _ = Xobl_bindings.X.open_display (Some ":0") in
     print_endline (Xobl_bindings.Protocol.show_handshake_resp resp);
     Lwt.return ())
