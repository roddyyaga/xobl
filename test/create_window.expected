{ Protocol.protocol_version = (11, 0); release_number = 12008000l;
  resource_id_base = 58720256l; resource_id_mask = 2097151l;
  motion_buffer_size = 256l; maximum_request_length = 65535;
  image_byte_order = Protocol.Little_endian;
  bitmap_format_bit_order = Protocol.Least_significant;
  bitmap_format_scanline_unit = 32; bitmap_format_scanline_pad = 32;
  keycode_range = { Protocol.min = 8; max = 255 };
  vendor = "The X.Org Foundation";
  pixmap_formats =
  [{ Protocol.pixmap_depth = 1; bits_per_pixel = 1; scanline_pad = 32 };
    { Protocol.pixmap_depth = 4; bits_per_pixel = 8; scanline_pad = 32 };
    { Protocol.pixmap_depth = 8; bits_per_pixel = 8; scanline_pad = 32 };
    { Protocol.pixmap_depth = 15; bits_per_pixel = 16; scanline_pad = 32 };
    { Protocol.pixmap_depth = 16; bits_per_pixel = 16; scanline_pad = 32 };
    { Protocol.pixmap_depth = 24; bits_per_pixel = 32; scanline_pad = 32 };
    { Protocol.pixmap_depth = 32; bits_per_pixel = 32; scanline_pad = 32 }];
  screens =
  [{ Protocol.root = 315l; default_colormap = 32l; white_pixel = 16777215l;
     black_pixel = 0l; current_input_masks = 5898292l;
     width_in_pixels = 1366; height_in_pixels = 768;
     width_in_millimeters = 361; height_in_millimeters = 203;
     installed_maps = { Protocol.min = 1; max = 1 }; root_visual = 33l;
     backing_stores = Protocol.Backing_stores_when_mapped;
     save_unders = false; root_depth = 24;
     depths =
     [{ Protocol.depth = 24;
        visual_types =
        [{ Protocol.visual_id = 33l;
           visual_class = Protocol.Visual_true_color; bits_per_rgb_value = 8;
           colormap_entries = 256; red_mask = 16711680l; green_mask = 65280l;
           blue_mask = 255l };
          { Protocol.visual_id = 34l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 243l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 244l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 245l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 246l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 247l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 248l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 249l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 250l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 251l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 252l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 253l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 254l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 255l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 256l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 257l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 258l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 259l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 260l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 261l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 262l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 263l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 264l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 265l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 266l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 267l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 268l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 269l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 270l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 271l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 272l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 273l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 274l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 275l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 276l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 277l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 278l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 279l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 280l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 281l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 282l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 283l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 284l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 285l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 286l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 287l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 288l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 289l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 290l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 291l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 292l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 293l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 294l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 295l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 296l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 297l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 298l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 299l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 300l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 301l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 302l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 303l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
          { Protocol.visual_id = 304l;
            visual_class = Protocol.Visual_direct_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l }
          ]
        };
       { Protocol.depth = 1; visual_types = [] };
       { Protocol.depth = 4; visual_types = [] };
       { Protocol.depth = 8; visual_types = [] };
       { Protocol.depth = 15; visual_types = [] };
       { Protocol.depth = 16; visual_types = [] };
       { Protocol.depth = 32;
         visual_types =
         [{ Protocol.visual_id = 152l;
            visual_class = Protocol.Visual_true_color;
            bits_per_rgb_value = 8; colormap_entries = 256;
            red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
           { Protocol.visual_id = 305l;
             visual_class = Protocol.Visual_true_color;
             bits_per_rgb_value = 8; colormap_entries = 256;
             red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
           { Protocol.visual_id = 306l;
             visual_class = Protocol.Visual_true_color;
             bits_per_rgb_value = 8; colormap_entries = 256;
             red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
           { Protocol.visual_id = 307l;
             visual_class = Protocol.Visual_true_color;
             bits_per_rgb_value = 8; colormap_entries = 256;
             red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
           { Protocol.visual_id = 308l;
             visual_class = Protocol.Visual_true_color;
             bits_per_rgb_value = 8; colormap_entries = 256;
             red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
           { Protocol.visual_id = 309l;
             visual_class = Protocol.Visual_true_color;
             bits_per_rgb_value = 8; colormap_entries = 256;
             red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
           { Protocol.visual_id = 310l;
             visual_class = Protocol.Visual_true_color;
             bits_per_rgb_value = 8; colormap_entries = 256;
             red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
           { Protocol.visual_id = 311l;
             visual_class = Protocol.Visual_true_color;
             bits_per_rgb_value = 8; colormap_entries = 256;
             red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
           { Protocol.visual_id = 312l;
             visual_class = Protocol.Visual_true_color;
             bits_per_rgb_value = 8; colormap_entries = 256;
             red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l };
           { Protocol.visual_id = 313l;
             visual_class = Protocol.Visual_true_color;
             bits_per_rgb_value = 8; colormap_entries = 256;
             red_mask = 16711680l; green_mask = 65280l; blue_mask = 255l }
           ]
         }
       ]
     }
    ]
  }
